test/unit/org/apache/cassandra/index/sai/analyzer/AnalyzerEqOperatorSupportTest.java

```diff
@@ -0,0 +1,368 @@
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'Quick' AND v = 'fox'");
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'quick' AND v = 'fox'");
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'Quick' AND v = 'Lazy'");
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'quick' AND v = 'lazy'");
+    }
+
+    private void assertNonTokenizedIndexSupportsMatches()
+    {
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'Quick fox'", row(1));
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'quick fox'", row(1));
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'fox'");
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'Quick fox' OR v : 'Lazy fox'", row(1), row(2));
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'quick fox' OR v : 'lazy fox'", row(1), row(2));
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'Quick' OR v : 'Lazy'");
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'quick' OR v : 'lazy'");
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'Quick' AND v : 'fox'");
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'quick' AND v : 'fox'");
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'Quick' AND v : 'Lazy'");
+        assertRowsWithoutWarning("SELECT k FROM %s WHERE v : 'quick' AND v : 'lazy'");
+    }
+
+    private void assertNonTokenizedIndexSupportsMixedEqualityAndMatches()
+    {
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'Quick fox' OR v : 'Lazy fox'", row(1), row(2));
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'quick fox' OR v : 'lazy fox'", row(1), row(2));
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'Quick' OR v : 'Lazy'");
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'quick' OR v : 'lazy'");
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'Quick' AND v : 'fox'");
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'quick' AND v : 'fox'");
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'Quick' AND v : 'Lazy'");
+        assertRowsWithWarning("SELECT k FROM %s WHERE v = 'quick' AND v : 'lazy'");
+
+        assertRowsWithWarning("SELECT k FROM %s WHERE v : 'Quick fox' OR v = 'Lazy fox'", row(1), row(2));
+        assertRowsWithWarning("SELECT k FROM %s WHERE v : 'quick fox' OR v = 'lazy fox'", row(1), row(2));
+        assertRowsWithWarning("SELECT k FROM %s WHERE v : 'Quick' OR v = 'Lazy'");
+        assertRowsWithWarning("SELECT k FROM %s WHERE v : 'quick' OR v = 'lazy'");
+        assertRowsWithWarning("SELECT k FROM %s WHERE v : 'Quick' AND v = 'fox'");
+        assertRowsWithWarning("SELECT k FROM %s WHERE v : 'quick' AND v = 'fox'");
+        assertRowsWithWarning("SELECT k FROM %s WHERE v : 'Quick' AND v = 'Lazy'");
+        assertRowsWithWarning("SELECT k FROM %s WHERE v : 'quick' AND v = 'lazy'");
```

Comment:
I hope no one uses something like that in prod...ever! I am tempted to prohibit mixing... But maybe it was too complicated to prohibit it?

Reply:
Me neither, although what we don't want is for anyone to use `=` in an analyzed column, and indeed users will receive a warning about the `=` operator. 

We support:
```
SELECT k FROM %s WHERE v : 'quick' AND v : 'lazy';
```
By setting `'equals_behaviour_when_analyzed': 'MATCH'` we are making `=` entirely equivalent to `:`, so we also support:
```
SELECT k FROM %s WHERE v = 'quick' AND v = 'lazy';
```
If those two queries work and have the same meaning, because we have configured the operators to be equivalent, it seems consistent that mixed combinations of both work too:
```
SELECT k FROM %s WHERE v = 'quick' AND v : 'lazy';
SELECT k FROM %s WHERE v : 'quick' AND v = 'lazy';
```
It's a bit odd, but I think it only exposes the oddness of having two operators with the same meaning. 

I mean, `'equals_behaviour_when_analyzed': 'MATCH'` works as if translating `=` on analyzed columns to `:`, which is what we expect users to do from now on. From the point of view of that configured behaviour, I think it's consistent to accept queries mixing both operators, which are considered identical. 

wdyt?

Reply:
While you are not wrong, it seems we were prohibiting mixing before.
`testAnalyzerMatchesAndEqualityFailForConjunction` was removed. I guess it was easy to prohibit as simply `=` for analyzers was broken. In that case I am fine to leave it as is. Maybe just add on commit one more small case where we use `'equals_behaviour_when_analyzed': 'UNSUPPORTED'` and mixed query. It will also save us from future regessions.

Reply:
I think we can do one of each, and not support mixed `:` and `=` if the analyzer doesn't support `=`. But if it does we should support both, and multiple `=` too. wdyt?

I have updated the test in that direction. Unfortunately, one of the cases is affected by [CNDB-10731](https://github.com/riptano/cndb/issues/10731), so I have left it commented with a `TODO` reminder.